sw/device/silicon_creator/mask_rom/mask_rom_epmp.c

```diff
@@ -71,3 +87,33 @@ void mask_rom_epmp_unlock_rom_ext_rx(epmp_state_t *state, epmp_region_t image) {
   CSR_CLEAR_BITS(CSR_REG_PMPCFG1, 0xff);
   CSR_SET_BITS(CSR_REG_PMPCFG1, kEpmpModeTor | kEpmpPermLockedReadExecute);
 }
+
+void mask_rom_epmp_config_debug_rom(lifecycle_state_t lc_state) {
+  const uint32_t pmpaddr = (TOP_EARLGREY_RV_DM_ROM_BASE_ADDR >> 2) |
+                           ((TOP_EARLGREY_RV_DM_ROM_SIZE_BYTES - 1) >> 3);
+  // Update the hardware configuration (CSRs).
+  //
+  // Entry is hardcoded as 13. Make sure to modify hardcoded values if changing
+  // kEntry.
+  //
+  // The `pmp13cfg` configuration is the second field in `pmpcfg3`.
+  //
+  //            32          24          16           8           0
+  //             +------------+------------+------------+------------+
+  // `pmpcfg3` = | `pmp15cfg` | `pmp14cfg` | `pmp13cfg` | `pmp12cfg` |
+  //             +------------+------------+------------+------------+
+  CSR_WRITE(CSR_REG_PMPADDR13, pmpaddr);
+  CSR_CLEAR_BITS(CSR_REG_PMPCFG3, 0xff00);
+  uint32_t pmpcfg;
+  switch (lc_state) {
+    case kLcStateTest:
+    case kLcStateDev:
+    case kLcStateRma:
+      pmpcfg = (kEpmpModeNapot | kEpmpPermLockedReadWriteExecute) << 8;
+      break;
+    default:
+      pmpcfg = (kEpmpModeNapot | kEpmpPermLockedNoAccess) << 8;
+      break;
+  }
```

Comment:
Please see my comment on the previous switch statement.

Reply:
Also, debug region is unlocked unconditionally in `mask_rom_epmp.S`, should this function be only responsible for locking it? Or should we handle both cases only here. @mundaym, any thoughts?

Reply:
I had to unlock the region in mask_rom_epmp.S so the debugger could stop at main() (Or anywhere in between) without knowing LC state.  This function updates the access permissions and locks it based on LCS.